src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,340 @@
+        }
+        
+    }
+
+    private boolean writeManifestAndCopyToArm(UserAccountEntity userAccount, ArmBatchItems batchItems, File archiveRecordsFile) {
+        try {
+            if (!batchItems.getSuccessful().isEmpty()) {
+                dataStoreToArmHelper.writeManifestFile(batchItems, archiveRecordsFile);
+                updateObjectStateRecordManifestSuccessOrFailure(batchItems, archiveRecordsFile);
+                copyMetadataToArm(archiveRecordsFile);
+            } else {
+                log.warn("No EODs were able to be processed, skipping manifest file creation");
+            }
+        } catch (Exception e) {
+            String errorMessage = String.format("Error during generation of DETS batch manifest file %s", archiveRecordsFile.getName());
+            log.error(errorMessage, e);
+            batchItems.getSuccessful().forEach(batchItem -> dataStoreToArmHelper.recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount));
+            final String errorMessageWithStackTrace = errorMessage + " - " + ExceptionUtils.getStackTrace(e);
+            batchItems.getFailed().forEach(batchItem -> updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessageWithStackTrace));
+            return false;
+        }
+        return true;
+    }
+
+    private String getManifestFilePrefix() {
+        return detsToArmProcessorConfiguration.getManifestFilePrefix();
+    }
+
+    private void updateObjectStateRecordManifestSuccessOrFailure(ArmBatchItems batchItems, File archiveRecordsFile) {
+        for (var batchItem : batchItems.getSuccessful()) {
+            ObjectStateRecordEntity objectStateRecord = getObjectStateRecordEntity(batchItem.getArmEod());
+            if (nonNull(objectStateRecord)) {
+                objectStateRecord.setFlagFileMfstCreated(true);
+                objectStateRecord.setDateFileMfstCreated(currentTimeHelper.currentOffsetDateTime());
+                objectStateRecord.setIdManifestFile(archiveRecordsFile.getName());
+                objectStateRecord.setFlagMfstTransfToArml(true);
+                objectStateRecord.setDateMfstTransfToArml(currentTimeHelper.currentOffsetDateTime());
+                objectStateRecordRepository.save(objectStateRecord);
+            }
+
+        }
+        for (var batchItem : batchItems.getFailed()) {
+            String errorMessage = "Manifest file creation failed";
+            updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessage);
+        }
+
+    }
+
+    private void updateObjectStateRecordStatus(ExternalObjectDirectoryEntity externalObjectDirectory, String errorMessage) {
+        ObjectStateRecordEntity objectStateRecord = getObjectStateRecordEntity(externalObjectDirectory);
+        if (nonNull(objectStateRecord)) {
+            if (nonNull(objectStateRecord.getObjectStatus())) {
+                objectStateRecord.setObjectStatus(objectStateRecord.getObjectStatus() + " " + errorMessage);
```

Comment:
if we append don't we have the issue that we would be preserving "old" errors that happened on previous retries? Those would be no longer relevant in the next retries

Reply:
yes when I added this it was because I wanted to make sure the cascading would pick up all errors as its possible to get more than one error but primarily it would only happen if something went wrong and then when its trying to clean up something else goes wrong. This was what I was referring to on your ticket about appending.